service/controller/v12/context/context.go

```diff
@@ -14,24 +14,29 @@ import (
 
 type contextKey string
 
-const serviceKey contextKey = "service"
+const controllerKey contextKey = "controller"
 
 type Context struct {
 	AWSClient      awsclient.Clients
 	AWSService     awsservice.Interface
 	CloudConfig    cloudconfig.Interface
 	CloudFormation cloudformationservice.CloudFormation
 	EBSService     ebs.Interface
+
+	// Status is holds the data used to communicate between controller's
+	// resources. It can be edited in place as Context is stored as
+	// a pointer withing context.Context.
+	Status Status
```

Comment:
I propose communicating via Status part of the Context. Something similar to CR.